src/scenes/spaceball.inc.c

```diff
@@ -1,63 +1,481 @@
-#include "asm/scenes/spaceball/asm_0801fc44.s"
+#include "global.h"
+#include "src/code_08001360.h"
+#include "src/code_08007468.h"
+#include "src/code_0800b778.h"
+#include "src/lib_0804c870.h"
 
-#include "asm/scenes/spaceball/asm_0801fcb0.s"
+// For readability.
+#define gSpaceballInfo D_030055d0->gameInfo.spaceball
 
-#include "asm/scenes/spaceball/asm_0801fd1c.s"
+// Game-Specific Macros:
+#define SPACEBALL_LAUNCHED 0
+#define SPACEBALL_HIT 1
+#define SPACEBALL_BARELY 2
```

Comment:
These names are also too vague, should label what kind of data they're supposed to be since I can't really tell